### DISCUSSION TITLE
dagster and ipython

### DISCUSSION THREAD

**Discussion by dridk** [dagster-io/dagster#19579] [Ideas]
hi, 

I would like to manipulate dagster from ipython. It would be helpful to play with dagster instance and understand how dagster works under the hood. 

For exemple , from the cli : 

```
dagster shell 

```

**Comment by dridk** [comment]
For example, I already success to wipe asset from ipython using : 

os.environ["DAGSTER_HOME"] = xxxxx
DagsterInstance.get().wipe(...)

That's awesome because I don't need to use the UI nor the the cli. 

I now wonder if I can materialize an asset, or add dynamically an asset.  How can I access to the code location definition?

**Reply by alangenfeld** [reply]
> How can I access to the code location definition?

There is no persisted representation of your code location (outside of dagster-cloud). So your code location definition is just the config file or cli flags you use https://docs.dagster.io/concepts/code-locations#deploying-and-loading-code-locations


In the context of ipython it will be easiest to interact with your Dagster definitions directly by importing them in and and using the python APIs like `materialize` https://docs.dagster.io/_apidocs/execution#dagster.materialize


Technically you could attempt to interact with the Dagster definitions via a code server like the webserver and daemon do, but there are not really public APIs for that. You could reference the codebase if this is something you wish to explore. 
https://docs.dagster.io/deployment/overview

**Reply by dridk** [reply]
If I use dagster.materialize() from ipython, will the UI update ? 

I means that there must be a python function that is the equivalent of clicking on the materialse button from dagit?.

**Reply by dridk** [reply]
It works !! Thanks !!! 

```
materialize(assets, instance=DagsterInstance.get() ) 
```
